pdns/dnsdist-web.cc

```diff
@@ -161,6 +161,11 @@ static bool isMethodAllowed(const YaHTTP::Request& req)
       return true;
     }
   }
+  if (req.method == "POST" && g_apiReadWrite) {
```

Comment:
Maybe you can add two version of lua hook? One for R/O API mode and one more for R/W API mode flag?